subprojects/gdk_rust/tests/integration.rs

```diff
@@ -119,3 +126,189 @@ fn liquid() {
 
     test_session.stop();
 }
+
+// Test the low-level spv_cross_validate()
+#[test]
+fn spv_cross_validate() {
+    // Scenario 1: our local chain is a minority fork
+    {
+        // Setup two competing chain forks at height 126 and 1142
+        let (mut test_session1, mut test_session2) = setup_forking_sessions(false);
+        test_session1.node_generate(5); // session1 is on a minority fork
+        test_session2.node_generate(1020); // session2 is on the most-work chain
+        test_session1.wait_block_status_change();
+        test_session2.wait_block_status_change();
+        assert_eq!(test_session1.session.block_status().unwrap().0, 126);
+        assert_eq!(test_session2.session.block_status().unwrap().0, 1141);
+
+        // Grab direct access to session1's HeadersChain
+        let session1_chain = get_chain(&mut test_session1);
+        assert_eq!(session1_chain.height(), 126);
+
+        // Cross-validate session1's chain against session'2 electrum server
+        let session2_electrum_url = ElectrumUrl::Plaintext(test_session2.electrs_url.clone());
+        let result = spv::spv_cross_validate(
+            &session1_chain,
+            &session1_chain.tip().block_hash(),
+            &session2_electrum_url,
+        )
+        .unwrap();
+
+        let inv = assert_unwrap_invalid(result);
+        assert_eq!(inv.common_ancestor, 121);
+        assert_eq!(inv.longest_height, 1141);
+
+        test_session2.stop();
+    }
+
+    // Scenario 2: our local chain is lagging behind a longer chain
+    {
+        // Setup two nodes, make session2 ahead by 12 blocks
+        let (mut test_session1, mut test_session2) = setup_forking_sessions(false);
+        test_session2.node_generate(12);
+        test_session2.wait_block_status_change();
+
+        // Grab direct access to session1's HeadersChain
+        let session1_chain = get_chain(&mut test_session1);
+        assert_eq!(session1_chain.height(), 121);
+
+        // Cross-validate session1's chain against session'2 electrum server
+        let session2_electrum_url = ElectrumUrl::Plaintext(test_session2.electrs_url.clone());
+        let result = spv::spv_cross_validate(
+            &session1_chain,
+            &session1_chain.tip().block_hash(),
+            &session2_electrum_url,
+        )
+        .unwrap();
+
+        assert_eq!(assert_unwrap_invalid(result).longest_height, 133);
+
+        test_session2.stop();
+    }
+}
+
+// Test high-level session management, background validation and transaction status
+#[test]
+fn spv_cross_validation_session() {
+    let (mut test_session1, test_session2) = setup_forking_sessions(true);
```

Comment:
```suggestion
    let (mut test_session1, mut test_session2) = setup_forking_sessions(true);
```
this is needed to stop the session, I forgot in the previous suggestion

Reply:
Fixed (squashed into the previous commit, now https://github.com/Blockstream/gdk/pull/59/commits/59b365494798491f3b7d103a2da7587ef219c84a)